src/plugins/intel_gpu/src/plugin/infer_request.cpp

```diff
@@ -188,6 +203,8 @@ Blob::Ptr InferRequest::GetBlob(const std::string& name) {
     InputInfo::Ptr foundInput;
     DataPtr foundOutput;
     bool is_input = findInputAndOutputBlobByName(name, foundInput, foundOutput);
+    auto node = is_input ? inputNodeByName(name) : outputNodeByName(name);
```

Comment:
As I see, here you added the same code as will be added to the infer request base class in this PR: https://github.com/openvinotoolkit/openvino/pull/9557

Maybe will be better to cherry-pick some changes from PR 9557?

Reply:
The code in PR 9557 works OK for inputs only. For outputs, it fails to locate the node with a given name (because result node and old-style output names mismatch), and gives false negative. So, I had to use another approach.

Reply:
Changed the code to use your methods from base class